motoman_driver/src/joint_trajectory_streamer.cpp

```diff
@@ -403,6 +438,7 @@ void MotomanJointTrajectoryStreamer::streamingThread()
       continue;
     }
 
+    // this does not lock smpl_msg_conx_mutex_, but the mutex from JointTrajectoryStreamer
     this->mutex_.lock();
```

Comment:
It could be nice, but that would require changes to `JointTrajectoryStreamer` in `industrial_robot_client` ([here](https://github.com/ros-industrial/industrial_core/blob/ea3d805e1271550be312409d6474177e67796e15/industrial_robot_client/include/industrial_robot_client/joint_trajectory_streamer.h#L120)), it's not part of `motoman_driver`.

Reply:
To clarify: I'm not against updating `industrial_robot_client`, but that would need additional testing, further delaying this PR.